### PR TITLE
Update CSL JSON schemas to draft-07

### PR DESCRIPTION
**Pull request by rmzelle**
Work in progress (planning to add commits to update schemas to latest draft-07)

(benefited a lot from https://jsonschemalint.com/ and https://json-schema.org/understanding-json-schema/)

### CODE REVIEW

**Comment by dhimmel** [comment]
From @rmzelle at https://github.com/citation-style-language/schema/pull/161#issuecomment-485223687:

> I prepared a minimal update to draft 6 (it looks like no changes are necessary for draft 7) at #168, inspired by this PR. I would prefer to merge that, and then create one or more additional PRs to pull in improvements/changes proposed here (such as using proper definitions etc.).

I agree it makes sense to perform the most minimal changes first in order to comply with the new draft. I'll review this PR. Hopefully, then afterwards, some of @dsifford's more substantive proposed changes in https://github.com/citation-style-language/schema/pull/161 can be split up into focused PRs.

File: tests/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-jsonref
-jsonschema
+jsonschema>=3.0.0a3
```
Comment (dhimmel) [inline tests/requirements.txt]: It looks like `jsonschema` finally has [v3 production releases](https://pypi.org/project/jsonschema/#history). Hence, I would change this line to `jsonschema>=3.0`.

**Comment by dsifford** [comment]
@dhimmel I reverted potentially breaking changes in #161 last night. So that one is prob good to go pending another once-over. 

See also: https://github.com/citation-style-language/schema/pull/161#discussion_r247332076

**Comment by dhimmel** [comment]
@rmzelle if this fully upgrades to draft-06, would it make sense to merge and then work on draft-07 in a subsequent PR?

Are you waiting on any additional testing? Anything I can do to help speed this PR up?

**Comment by bdarcus** [comment]
@dhimmel - I was just wondering this. Do have thoughts on how to proceed at this point?

I know @rmzelle is really busy with other things these days.

**Comment by bdarcus** [comment]
Of course, latest json schema version is 2019-09.

Is there reason for us to maintain multiple versions, or should be just jump to latest?

**Comment by dhimmel** [comment]
> Do have thoughts on how to proceed at this point?

@rmzelle are you waiting on anything to merge. This PR is good to go right?

> Is there reason for us to maintain multiple versions, or should we just jump to the latest?

Doesn't seem like there's the developer capacity to support multiple schema versions. I think it makes sense to support a single schema version. Whatever is stable enough to be widely supported and not so old as to be deprecated or lacking in features. Based on https://json-schema.org/implementations.html, it looks like draft-07 would be the ideal version currently.

Upgrading to draft-06 is a step in the right direction.

**Comment by bdarcus** [comment]
I also am unclear how this PR relates to this other one:

https://github.com/citation-style-language/schema/pull/161

**Comment by rmzelle** [comment]
> I also am unclear how this PR relates to this other one:
> 
> #161

Derek upgraded the schema and made a bunch of schema changes within the same pull request, so I decided to make my own pull requests with just schema version upgrades. If I recall correctly, there are no differences between draft-06 and draft-07 (at least to the extent that they would affect our schema), and this might be good to merge, but I'm not 100% sure of that.

**Comment by bdarcus** [comment]
So idea is this would be merged first, and then #161 could be merged after?

I guess at this point we don't really need a version fro draft-06 though;
right? So any reason not to just jump directly to a draft-07 version?

>

**Comment by rmzelle** [comment]
> So idea is this would be merged first, and then #161 could be merged after? I guess at this point we don't really need a version fro draft-06 though; right? So any reason not to just jump directly to a draft-07 version?

It would replace #161. There were a lot of proposed changes in #161 that really should be split out into smaller PRs (like tightening the schema which might invalidate existing CSL JSON).

And it's been a long time since I looked at this, but if draft-06 and draft-07 are indeed identical in terms of schema, we can probably just change `"$schema" : "http://json-schema.org/draft-06/schema#"` to `"$schema" : "http://json-schema.org/draft-07/schema#"`, yes.

**Comment by bdarcus** [comment]
OIC. So ideally we'd test out your changes here to confirm they work correctly, and if good, merge, then update to the draft-07, and close #161 without merging?

**Comment by rmzelle** [comment]
Right. I'm sure there are good suggestions in #161, but the scope is a bit unmanageable, and it will have merge conflicts with this PR, so I would just split out any proposed changes there into fresh PRs once this one has been merged.

**Comment by bdarcus** [comment]
OK, I just updated to draft-07 here. 

@dhimmel - any chance you can test this and confirm it's good to merge from your end?

**Comment by dhimmel** [comment]
> OK, I just updated to draft-07 here.

Awesome.

> can test this and confirm it's good to merge from your end?

It's good on my end if the test suite passes. I wonder why travis CI doesn't appear to be running on https://github.com/citation-style-language/schema/pull/168/commits/f3ea4bf7dba472619d50256d48398b2d1632e0a3

**Comment by rmzelle** [comment]
> I wonder why travis CI doesn't appear to be running on [f3ea4bf](https://github.com/citation-style-language/schema/commit/f3ea4bf7dba472619d50256d48398b2d1632e0a3)

It ran: https://travis-ci.org/github/citation-style-language/schema/builds/689785207

**Comment by bdarcus** [comment]
Is there any note to squash them into single 07 commit?

**Comment by dhimmel** [comment]
> It ran: https://travis-ci.org/github/citation-style-language/schema/builds/689785207

Great output looks good. Usually a green check appears but that's missing. 

> Is there any not to squash them into a single 07 commit?

you can do this with a squash merge, although it will collapse the authorship since multiple people authored commits here. Not sure if you care.

File: tests/requirements.txt (hunk 1 above)
Comment (dhimmel) [reply tests/requirements.txt]: thanks @bdarcus for addressing in https://github.com/citation-style-language/schema/pull/168/commits/26cccd0e67759a073af7a9e401b1c5cdf9025855. This shouldn't actually change anything functionally but looks a bit cleaner.

**Comment by bdarcus** [comment]
Done. I added a text note to attribute rintze, but turns out wasn't necessary. My changes were trivial, so I don't care.

We should be good to go then. Thanks you two!

**Pull request merged**